tests/ServiceProviderTest.php

```diff
@@ -37,6 +45,163 @@ public function testBugsnagLoggerIsInjectable()
+        $expectedStripPathRegex = $expectedProjectRootRegex;
+
+        $this->assertSame(
+            $expectedStripPathRegex,
+            $stripPathRegex,
+            "Expected to set a sensible default for the 'stripPathRegex'"
+        );
+        $this->assertSame(
+            $expectedProjectRootRegex,
+            $projectRootRegex,
+            "Expected to set a sensible default for the 'projectRootRegex'"
+        );
+    }
+
+    /**
+     * @param string|null $projectRoot
+     * @param string|null $stripPath
+     * @param string|null $expectedProjectRootRegex
+     * @param string      $expectedStripPathRegex
+     *
+     * @return void
+     *
+     * @dataProvider projectRootAndStripPathProvider
+     */
+    public function testProjectRootAndStripPathAreSetCorrectly($projectRoot, $stripPath, $expectedProjectRootRegex, $expectedStripPathRegex)
+    {
+        /** @var \Illuminate\Config\Repository $laravelConfig */
+        $laravelConfig = $this->app->config;
+        $bugsnagConfig = $laravelConfig->get('bugsnag');
+
+        $this->assertNull(
+            $bugsnagConfig['project_root'],
+            "Expected the default configuration value for 'project_root' to be null"
+        );
+
+        $this->assertNull(
+            $bugsnagConfig['strip_path'],
+            "Expected the default configuration value for 'strip_path' to be null"
+        );
+
+        $bugsnagConfig['project_root'] = $projectRoot;
+        $bugsnagConfig['strip_path'] = $stripPath;
+
+        $laravelConfig->set('bugsnag', $bugsnagConfig);
+
+        $client = $this->app->make(Client::class);
+
+        $this->assertInstanceOf(Client::class, $client);
+
+        $basePath = $this->app->basePath();
+        $appRoot = $this->app->path();
+
+        /** @var Client $client */
+        $config = $client->getConfig();
+
+        $projectRootRegex = $this->getProperty($config, 'projectRootRegex');
+        $stripPathRegex = $this->getProperty($config, 'stripPathRegex');
+
+        $this->assertSame(
+            $expectedStripPathRegex,
+            $stripPathRegex,
+            "Expected the 'stripPathRegex' to match the string provided in Bugsnag configuration"
+        );
+
+        $this->assertSame(
+            $expectedProjectRootRegex,
+            $projectRootRegex,
+            "Expected the 'projectRootRegex' to match the string provided in Bugsnag configuration"
+        );
+    }
+
+    public function projectRootAndStripPathProvider()
+    {
+        return [
+            // If both parameters are provided, the project root should be null
+            // and the strip path set to a regex matching the given string
+            // TODO this might be a bug — when a strip path value is configured,
+            //      we only set the project root path if one wasn't provided. If both
+            //      strip path _and_ project root are given, no project root is set
```

Comment:
I think this is a bug, but it still needs investigation to see what the consequences are (relevant code here](https://github.com/bugsnag/bugsnag-laravel/blob/master/src/BugsnagServiceProvider.php#L341-L349))